mesheryctl/pkg/utils/healthcheck_test.go

```diff
@@ -0,0 +1,124 @@
+package utils
+
+import (
+	"io/ioutil"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/version"
+)
+
+var versionCheck = []struct {
+	version  string // input
+	expected [3]int // expected output
+}{
+	{"1.12.0", [3]int{1, 12, 0}},
+	{"1.12.1", [3]int{1, 12, 1}},
+	{"1.12.2", [3]int{1, 12, 2}},
+	{"1.12.3", [3]int{1, 12, 3}},
+	{"1.12.4", [3]int{1, 12, 4}},
+	{"1.12.5", [3]int{1, 12, 5}},
+	{"1.12.6", [3]int{1, 12, 6}},
+	{"1.12.7", [3]int{1, 12, 7}},
+	{"1.12.8", [3]int{1, 12, 8}},
+	{"1.12.9", [3]int{1, 12, 9}},
+	{"1.12.10", [3]int{1, 12, 10}},
+}
+
+func TestGetK8sVersionInfo(t *testing.T) {
+	tests := []struct {
+		version  string        // input
+		expected *version.Info // expected output
+		fixture  string        // fixture file
+	}{
+		{
+			version:  "v1.12.0",
+			expected: &version.Info{Major: "1", Minor: "12", GitVersion: "1.12.0-alpha.1+1"},
+			fixture:  "k8s_version_info.golden",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.version, func(t *testing.T) {
+			_, err := ioutil.ReadFile("fixtures/healthcheck/" + tt.fixture)
+			if err != nil {
+				t.Fatalf("Error reading fixture file: %v", err)
+			}
+		})
+	}
+}
+
+func TestCheckK8sVersion(t *testing.T) {
```

Comment:
Better renaming the test function, because the name doesn't match the function it's testing.